receiver/datadogreceiver/tags_test.go

```diff
@@ -0,0 +1,160 @@
+				"host.name": "host",
+			}),
+			expectedScopeAttrs: pcommon.NewMap(),
+			expectedDpAttrs:    pcommon.NewMap(),
+		},
+		{
+			name: "provides both host and tags where some tag keys have to replaced by otel conventions",
+			tags: []string{"env:prod", "service:my-service", "version:1.0"},
+			host: "host",
+			expectedResourceAttrs: newMapFromKV(t, map[string]any{
+				"host.name":              "host",
+				"deployment.environment": "prod",
+				"service.name":           "my-service",
+				"service.version":        "1.0",
+			}),
+			expectedScopeAttrs: pcommon.NewMap(),
+			expectedDpAttrs:    pcommon.NewMap(),
+		},
+		{
+			name: "provides host, tags and unnamed tags",
+			tags: []string{"env:prod", "foo"},
+			host: "host",
+			expectedResourceAttrs: newMapFromKV(t, map[string]any{
+				"host.name":              "host",
+				"deployment.environment": "prod",
+			}),
+			expectedScopeAttrs: pcommon.NewMap(),
+			expectedDpAttrs: newMapFromKV(t, map[string]any{
+				"unnamed_foo": "foo",
+			}),
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			pool := newStringPool()
+			resourceAttrs, scopeAttrs, dpAttrs := tagsToAttributes(c.tags, c.host, pool)
+
+			assert.Equal(t, c.expectedResourceAttrs.Len(), resourceAttrs.Len())
+			c.expectedResourceAttrs.Range(func(k string, _ pcommon.Value) bool {
+				ev, _ := c.expectedResourceAttrs.Get(k)
+				av, ok := resourceAttrs.Get(k)
+				assert.True(t, ok)
+				assert.Equal(t, ev, av)
+				return true
+			})
+
+			assert.Equal(t, c.expectedScopeAttrs.Len(), scopeAttrs.Len())
+			c.expectedScopeAttrs.Range(func(k string, _ pcommon.Value) bool {
+				ev, _ := c.expectedScopeAttrs.Get(k)
+				av, ok := scopeAttrs.Get(k)
+				assert.True(t, ok)
+				assert.Equal(t, ev, av)
+				return true
+			})
+
+			assert.Equal(t, c.expectedDpAttrs.Len(), dpAttrs.Len())
+			c.expectedDpAttrs.Range(func(k string, _ pcommon.Value) bool {
+				ev, _ := c.expectedDpAttrs.Get(k)
+				av, ok := dpAttrs.Get(k)
+				assert.True(t, ok)
+				assert.Equal(t, ev, av)
+				return true
+			})
+		})
+
+	}
+
+}
+
+func newMapFromKV(t *testing.T, kv map[string]any) pcommon.Map {
+	m := pcommon.NewMap()
+	err := m.FromRaw(kv)
+	assert.NoError(t, err)
+	return m
+}
+
+func TestDatadogTagToKeyValuePair(t *testing.T) {
+	cases := []struct {
+		name          string
+		input         string
+		expectedKey   string
+		expectedValue string
+	}{
+		{
+			name:          "empty",
+			input:         "",
+			expectedKey:   "",
+			expectedValue: "",
+		},
+		{
+			name:          "kv tag",
+			input:         "foo:bar",
+			expectedKey:   "foo",
+			expectedValue: "bar",
+		},
+		{
+			name:          "unnamed tag",
+			input:         "foo",
+			expectedKey:   "unnamed_foo",
+			expectedValue: "foo",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			key, value := translateDatadogTagToKeyValuePair(c.input)
+			if key != c.expectedKey {
```

Comment:
I wonder why not use assert.Equal() here

Reply:
You are 100% right, just did!